SHOW STATS FOR return null stats when reading timestamp with/without timezone stats > 9999 year in Delta Lake

TIMESTAMP type:
```sql
trino> create table test(a timestamp) with (location = 's3://test-bucket/test');
spark> insert into test values (timestamp '10000-12-31T23:59:59.999999');
trino> table test; -- returns the expected result
trino> show stats for test;
```

TIMESTAMP WITH TIME ZONE type:
```sql
trino> create table testw(a timestamp with time zone) with (location = 's3://test-bucket/testw');
spark> insert into testw values (timestamp '10000-12-31T23:59:59.999999');
trino> table testw; -- returns the expected result
trino> show stats for testw;
```

The bottom SHOW STATS statements return NULL in low and high columns due to an internal error. 
```
trino:default> show stats for test;
 column_name | data_size | distinct_values_count | nulls_fraction | row_count | low_value | high_value
-------------+-----------+-----------------------+----------------+-----------+-----------+------------
 a           |      NULL |                  NULL |           NULL |      NULL | NULL      | NULL
 NULL        |      NULL |                  NULL |           NULL |      NULL | NULL      | NULL
```

The sever log prints the following error messages. Probably, we should replace `SignStyle.NORMAL` with `SignStyle.EXCEEDS_PAD` in TransactionLogParser class. Also, we should confirm the behavior on partition columns. 
```
presto-master       | io.trino.spi.TrinoException: Unable to parse value [+10000-12-31T23:59:59.999] from column a with type timestamp(6)
presto-master       | 	at io.trino.plugin.deltalake.transactionlog.TransactionLogParser.deserializeColumnValue(TransactionLogParser.java:226)
presto-master       | 	at io.trino.plugin.deltalake.transactionlog.statistics.DeltaLakeJsonFileStatistics.deserializeStatisticsValue(DeltaLakeJsonFileStatistics.java:144)
presto-master       | 	at io.trino.plugin.deltalake.transactionlog.statistics.DeltaLakeJsonFileStatistics.lambda$getMinColumnValue$4(DeltaLakeJsonFileStatistics.java:136)
presto-master       | 	at java.base/java.util.Optional.flatMap(Optional.java:289)
presto-master       | 	at io.trino.plugin.deltalake.transactionlog.statistics.DeltaLakeJsonFileStatistics.getMinColumnValue(DeltaLakeJsonFileStatistics.java:136)
presto-master       | 	at io.trino.plugin.deltalake.statistics.FileBasedTableStatisticsProvider.getTableStatistics(FileBasedTableStatisticsProvider.java:174)
presto-master       | 	at io.trino.plugin.deltalake.DeltaLakeMetadata.getTableStatistics(DeltaLakeMetadata.java:884)
presto-master       | 	at io.trino.plugin.base.classloader.ClassLoaderSafeConnectorMetadata.getTableStatistics(ClassLoaderSafeConnectorMetadata.java:353)
presto-master       | 	at io.trino.tracing.TracingConnectorMetadata.getTableStatistics(TracingConnectorMetadata.java:330)
presto-master       | 	at io.trino.metadata.MetadataManager.getTableStatistics(MetadataManager.java:476)
presto-master       | 	at io.trino.tracing.TracingMetadata.getTableStatistics(TracingMetadata.java:310)
presto-master       | 	at io.trino.cost.CachingTableStatsProvider.getTableStatistics(CachingTableStatsProvider.java:46)
presto-master       | 	at io.trino.cost.TableScanStatsRule.doCalculate(TableScanStatsRule.java:60)
presto-master       | 	at io.trino.cost.TableScanStatsRule.doCalculate(TableScanStatsRule.java:36)
presto-master       | 	at io.trino.cost.SimpleStatsRule.calculate(SimpleStatsRule.java:37)
presto-master       | 	at io.trino.cost.ComposableStatsCalculator.calculateStats(ComposableStatsCalculator.java:83)
presto-master       | 	at io.trino.cost.ComposableStatsCalculator.calculateStats(ComposableStatsCalculator.java:71)
presto-master       | 	at io.trino.cost.CachingStatsProvider.getStats(CachingStatsProvider.java:87)
presto-master       | 	at io.trino.cost.OutputStatsRule.calculate(OutputStatsRule.java:39)
presto-master       | 	at io.trino.cost.OutputStatsRule.calculate(OutputStatsRule.java:25)
presto-master       | 	at io.trino.cost.ComposableStatsCalculator.calculateStats(ComposableStatsCalculator.java:83)
presto-master       | 	at io.trino.cost.ComposableStatsCalculator.calculateStats(ComposableStatsCalculator.java:71)
presto-master       | 	at io.trino.cost.CachingStatsProvider.getStats(CachingStatsProvider.java:87)
presto-master       | 	at io.trino.cost.StatsAndCosts.create(StatsAndCosts.java:87)
presto-master       | 	at io.trino.sql.planner.LogicalPlanner.plan(LogicalPlanner.java:291)
presto-master       | 	at io.trino.sql.analyzer.QueryExplainer.getLogicalPlan(QueryExplainer.java:176)
presto-master       | 	at io.trino.sql.rewrite.ShowStatsRewrite$Visitor.visitShowStats(ShowStatsRewrite.java:150)
presto-master       | 	at io.trino.sql.rewrite.ShowStatsRewrite$Visitor.visitShowStats(ShowStatsRewrite.java:117)
presto-master       | 	at io.trino.sql.tree.ShowStats.accept(ShowStats.java:45)
presto-master       | 	at io.trino.sql.tree.AstVisitor.process(AstVisitor.java:27)
presto-master       | 	at io.trino.sql.rewrite.ShowStatsRewrite.rewrite(ShowStatsRewrite.java:114)
presto-master       | 	at io.trino.sql.rewrite.StatementRewrite.rewrite(StatementRewrite.java:54)
presto-master       | 	at io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java:92)
presto-master       | 	at io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java:86)
presto-master       | 	at io.trino.execution.SqlQueryExecution.analyze(SqlQueryExecution.java:285)
presto-master       | 	at io.trino.execution.SqlQueryExecution.<init>(SqlQueryExecution.java:218)
presto-master       | 	at io.trino.execution.SqlQueryExecution$SqlQueryExecutionFactory.createQueryExecution(SqlQueryExecution.java:884)
presto-master       | 	at io.trino.dispatcher.LocalDispatchQueryFactory.lambda$createDispatchQuery$0(LocalDispatchQueryFactory.java:153)
presto-master       | 	at io.trino.$gen.Trino_448_224_g569b045____20240529_090522_2.call(Unknown Source)
presto-master       | 	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:131)
presto-master       | 	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:76)
presto-master       | 	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:82)
presto-master       | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
presto-master       | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
presto-master       | 	at java.base/java.lang.Thread.run(Thread.java:1570)
presto-master       | Caused by: java.time.format.DateTimeParseException: Text '+10000-12-31T23:59:59.999' could not be parsed at index 0
presto-master       | 	at java.base/java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:2108)
presto-master       | 	at java.base/java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:2010)
presto-master       | 	at java.base/java.time.LocalDateTime.parse(LocalDateTime.java:494)
presto-master       | 	at io.trino.plugin.deltalake.transactionlog.statistics.DeltaLakeJsonFileStatistics.readStatisticsTimestamp(DeltaLakeJsonFileStatistics.java:172)
presto-master       | 	at io.trino.plugin.deltalake.transactionlog.TransactionLogParser.deserializeColumnValue(TransactionLogParser.java:214)
presto-master       | 	... 44 more
```

--- a/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogParser.java
+++ b/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogParser.java
@@ -119,6 +119,11 @@ public final class TransactionLogParser
             .withResolverStyle(ResolverStyle.STRICT);
     public static final DateTimeFormatter JSON_STATISTICS_TIMESTAMP_FORMATTER = new DateTimeFormatterBuilder()
             .parseCaseInsensitive()
+            // Using optional format because the format differ among Delta Lake versions.
+            // For instance, newer versions use '+' sign when year is more than 4 digits, but older versions don't write the sign.
+            .optionalStart()
+            .appendLiteral('+')
+            .optionalEnd()
             .appendValue(YEAR, 4, 10, SignStyle.NORMAL)
             .appendLiteral('-')
             .appendValue(MONTH_OF_YEAR, 2)

--- a/plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java
+++ b/plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java
@@ -784,6 +784,31 @@ public class TestDeltaLakeConnectorTest
     }
 
     @Test
+    public void testShowStatsForTimestampWithTimeZone()
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_stats_timestamptz_", "(x TIMESTAMP(3) WITH TIME ZONE) WITH (checkpoint_interval = 2)")) {
+            assertUpdate("INSERT INTO " + table.getName() + " VALUES (TIMESTAMP '+10000-01-02 13:34:56.123 +01:00')", 1);
+            assertThat(query("SHOW STATS FOR " + table.getName()))
+                    .result()
+                    .projected("column_name", "low_value", "high_value")
+                    .skippingTypesCheck()
+                    .matches("VALUES " +
+                            "('x', '+10000-01-02 12:34:56.123 UTC', '+10000-01-02 12:34:56.123 UTC')," +
+                            "(null, null, null)");
+
+            assertUpdate("INSERT INTO " + table.getName() + " VALUES (TIMESTAMP '-9999-01-02 13:34:56.123 +01:00')", 1);
+            assertThat(query("SHOW STATS FOR " + table.getName()))
+                    .result()
+                    .projected("column_name", "low_value", "high_value")
+                    .skippingTypesCheck()
+                    .matches("VALUES " +
+                            // The negative timestamp is discarded by TransactionLogParser.START_OF_MODERN_ERA_DATE
+                            "('x', '+10000-01-02 12:34:56.123 UTC', '+10000-01-02 12:34:56.123 UTC')," +
+                            "(null, null, null)");
+        }
+    }
+
+    @Test
     public void testAddColumnToPartitionedTable()
     {
         try (TestTable table = new TestTable(getQueryRunner()::execute, "test_add_column_partitioned_table_", "(x VARCHAR, part VARCHAR) WITH (partitioned_by = ARRAY['part'])")) {

